core/src/sql/value/value.rs

```diff
@@ -2245,9 +2245,9 @@ impl Value {
 
 	/// Try to convert this value to a Record of a certain type
 	pub(crate) fn convert_to_record_type(self, val: &[Table]) -> Result<Thing, Error> {
-		match self {
+		match self.clone().convert_to_record() {
 			// Records are allowed if correct type
-			Value::Thing(v) if self.is_record_type(val) => Ok(v),
+			Ok(t) if t.is_record_type(val) => Ok(t),
 			// Anything else raises an error
 			_ => Err(Error::ConvertTo {
 				from: self,
```

Comment:
You could avoid a clone here by changing it to

```rust
		match self.convert_to_record()? {
			// Records are allowed if correct type
			t if t.is_record_type(val) => Ok(t),
			// Anything else raises an error
			from => Err(Error::ConvertTo {
				from,
```

Reply:
Fixed in [0b7d146](https://github.com/surrealdb/surrealdb/pull/4496/commits/0b7d146f94658a96504c389e052b09199512c459)

Reply:
You do actually need to clone there, as we want to retain the original value in the error